modules/andes-core/broker/src/main/java/org/wso2/andes/store/FailureObservingMessageStore.java

```diff
@@ -437,6 +437,19 @@ public void addQueue(String storageQueueName) throws AndesException {
      * {@inheritDoc}
      */
     @Override
+    public Map<String, Integer> getMessageCountForAllQueues(List<String> queueNames) throws AndesException {
+        try {
+            return wrappedInstance.getMessageCountForAllQueues(queueNames);
+        } catch (AndesStoreUnavailableException exception) {
+            notifyFailures(exception);
```

Comment:
Can we introduce another method like notifyFailuresAndThrow so we don't have to have this sequence of statements redundantly?